comms/dht/src/peer_validator.rs

```diff
@@ -73,9 +76,23 @@ impl<'a> PeerValidator<'a> {
             });
         }
 
+        if let Some(existing) = &existing_peer {
```

Comment:
This isn't something that can happen (bar some peerdb coding error). The caller must provide the correct existing peer and loads the peer by their authenticated public key. So this is checking that the developer has done things correctly rather than a legitimate peer validation error. I would say the check is not needed, alternatively, you could comment that this is a coding bug, and perhaps prefix the error text with `BUG: `.